go/vt/vtgate/planbuilder/testdata/filter_cases.json

```diff
@@ -797,6 +797,79 @@
       ]
     }
   },
+  {
+    "comment": "Disjunction of conjunctions with 4 or more disjunctions",
+    "query": "select id from user where (col = 'aa' AND name = 'bb') OR (col = 'cc' AND name = 'dd') OR (col = 'ee' AND name = 'ff') OR (col = 'gg' AND name = 'hh')",
+    "plan": {
+      "QueryType": "SELECT",
+      "Original": "select id from user where (col = 'aa' AND name = 'bb') OR (col = 'cc' AND name = 'dd') OR (col = 'ee' AND name = 'ff') OR (col = 'gg' AND name = 'hh')",
+      "Instructions": {
+        "OperatorType": "Route",
+        "Variant": "Scatter",
+        "Keyspace": {
+          "Name": "user",
+          "Sharded": true
+        },
+        "FieldQuery": "select id from `user` where 1 != 1",
+        "Query": "select id from `user` where col = 'aa' and `name` = 'bb' or col = 'cc' and `name` = 'dd' or col = 'ee' and `name` = 'ff' or col = 'gg' and `name` = 'hh'",
+        "Table": "`user`"
+      },
+      "TablesUsed": [
+        "user.user"
+      ]
+    }
```

Comment:
These queries generated `MultiEqual` opcodes on v19 and later, but don't do that on v18. I had to decide on whether to remove these tests or to update them.

I decided to update them with the plans they generate on v18.